lib/cn/cn.c

```diff
@@ -478,21 +485,31 @@ cn_mblocks_destroy(
     u32 lx;
 
     for (lx = 0; lx < num_lists; lx++) {
-        cn_delete_blks(mp, &list[lx].kblks, n_committed);
-        if (kcompact)
-            continue;
-
-        cn_delete_blks(mp, &list[lx].vblks, n_committed);
+        /* The order of destroying here, hblock, kblocks, and then vblocks is
+         * sensitive due to the n_committed argument. This argument is passed
+         * from cn_mblocks_commit() to determine which blocks need to be aborted
+         * or deleted.
+         */
 
-       /* Same logic as cn_delete_blks() applied to one block. Candidate for
-        * refactor.
-        */
+        /* Same logic as cn_delete_blks() applied to one block. Candidate for
+         * refactor.
+         */
         if (n_committed > 0) {
             delete_mblock(mp, &list[lx].hblk);
+            n_committed--;
         } else {
             abort_mblock(mp, &list[lx].hblk);
         }
+
+        cn_delete_blks(mp, &list[lx].kblks, &n_committed);
+        if (kcompact)
```

Comment:
You can get rid of this continue by checking the inverse of this condition.

if (!kcompact)
    cn_delete_blks(...);

Reply:
Is one better than the other? I don't know!

Reply:
Just reads better IMO.